openbas-api/src/main/java/io/openbas/migration/V3_12__Add_index_for_atomic_testings.java

```diff
@@ -0,0 +1,26 @@
+package io.openbas.migration;
+
+import java.sql.Connection;
+import java.sql.Statement;
+import org.flywaydb.core.api.migration.BaseJavaMigration;
+import org.flywaydb.core.api.migration.Context;
+import org.springframework.stereotype.Component;
+
+@Component
+public class V3_12__Add_index_for_atomic_testings extends BaseJavaMigration {
+
+  @Override
+  public void migrate(Context context) throws Exception {
+    Connection connection = context.getConnection();
+    Statement select = connection.createStatement();
+    select.execute(""
+        + "CREATE INDEX idx_fk_inject_expectation ON injects_expectations (inject_id);"
+        + "CREATE INDEX idx_fk_inject_tag ON injects_tags (inject_id);"
+        + "CREATE INDEX idx_fk_inject_assets ON injects_assets (inject_id);"
+        + "CREATE INDEX idx_fk_inject_asset_groups ON injects_asset_groups (inject_id);"
+        + "CREATE INDEX idx_fk_inject_teams ON injects_teams (inject_id);"
+        + "CREATE INDEX idx_fk_inject_injects_documents ON injects_documents (inject_id);"
+        + "CREATE INDEX idx_null_exercise_and_scenario ON injects (inject_id) WHERE inject_scenario IS NULL AND inject_exercise IS NULL;"
```

Comment:
Why we need this index ?

Reply:
The last one is for atomic testing, where the condition to filter injects is that the scenario and exercise are null.

Reply:
Okay, but we already have an index on inject_id
How does this add improvement? (sorry but MY knowledge of indexes is limited)

Reply:
It depends on the type of search being performed. A search by injectId is different from a search by exercise and scenario... but, if the index is useful, it is the planificator who decides...